src/qt/veil/veilstatusbar.cpp

```diff
@@ -62,48 +70,55 @@ void VeilStatusBar::onBtnSyncClicked(){
 #ifdef ENABLE_WALLET
 bool fBlockNextStakeCheckSignal = false;
 void VeilStatusBar::setStakingText() {
+	stakingTextUpdateTimer->stop();
+
+	auto pwallet = GetMainWallet();
+
+	bool fStakingActive = pwallet->IsStakingActive();
 
-    // Determine if staking is recently active. Note that this is not immediate effect. Staking could be disabled and it could take some time (activeStakingMaxTime) update state.
-    int64_t nTimeLastHashing = 0;
-    if (!mapHashedBlocks.empty()) {
-        auto pindexBest = chainActive.Tip();
-        if (mapHashedBlocks.count(pindexBest->GetBlockHash())) {
-            nTimeLastHashing = mapHashedBlocks.at(pindexBest->GetBlockHash());
-        } else if (mapHashedBlocks.count(pindexBest->pprev->GetBlockHash())) {
-            nTimeLastHashing = mapHashedBlocks.at(pindexBest->pprev->GetBlockHash());
-        }
-    }
-    bool fStakingActive = false;
-    if (nTimeLastHashing)
-        fStakingActive = GetAdjustedTime() + MAX_FUTURE_BLOCK_TIME - nTimeLastHashing < ACTIVE_STAKING_MAX_TIME;
-	
     WalletModel::EncryptionStatus eStatus = this->walletModel->getEncryptionStatus();
 
     if (syncFlag){
-	ui->checkStaking->setText("Staking Disabled while Syncing");
+    	ui->checkStaking->setText("Staking Disabled while Syncing");
+    	ui->checkStaking->setStyleSheet(toggleOffGrey);
     }else if (WalletModel::Locked == eStatus) {
         ui->checkStaking->setText("Unlock wallet for Staking");
+        ui->checkStaking->setStyleSheet(toggleOffGrey);
     }else if (this->walletModel->isStakingEnabled()) {
-	if (fStakingActive) {
-            ui->checkStaking->setText("Staking Enabled");
-	}else{
-
-            interfaces::Wallet& wallet = walletModel->wallet();
-            interfaces::WalletBalances balances = wallet.getBalances();
-            int64_t zerocoin_balance = balances.zerocoin_balance;
-
-	    if (0.0 < zerocoin_balance) {
-                ui->checkStaking->setText("Enabling...");
-            }else{
-                ui->checkStaking->setText("You need some zerocoin");
-            }
-	}
+		if (fStakingActive) {
+				ui->checkStaking->setText("Staking Enabled");
+				ui->checkStaking->setStyleSheet(toggleOnBlue);
```

Comment:
This indentation is not only off the standard style, but it's off the local code indenting as well